src/components/inspector/SnapshotsDropdown.tsx

```diff
@@ -33,10 +35,10 @@ import { useEditorStore } from '../track-changes/useEditorStore'
 
 export const SnapshotsDropdown: React.FC = () => {
   const { wrapperRef, toggleOpen, isOpen } = useDropdown()
-  const snapshotStore = useSnapshotStore()
+  const [api] = useStore((store) => store.api)
```

Comment:
Exposing API to be something available through the store or to be available to components in general, looks to me like we overexpose it. It would be preferable if api was contained on its own architectural level and was abstracted away without being needed to components or hooks on every rerender. This is why we designed the Source strategy: we abstract APIs and data assembly away from react and prepare app to work with data in a concise and structured way.
One way we can solve this is to initiate stores on higher level (in EditorApp.tsx for example) and make sure we pass api there on initiation. But it seems to me, the most optimal solution would to to get rid of the zustand stores (snapshot store, doc store, editor state store) as a part of this PR and put all the api communication in one place.
In [LEAN-2702](https://github.com/Atypon-OpenSource/manuscripts-article-editor/pull/252/files#diff-c8f08befaeaf66733590edc1fa0590f40fa68b52f596f8ee5fdc03de32980aff) we are going to introduce a new data source for QB related data and this is something what we can take advantage of to make the put all the communcation scattered in the hook into one place.
This is especially considering that we do many changes here to those stores, while ideally they shouldn't really exist anyway.

Reply:
cc @mnatsheh

Reply:
I will wait for LEAN-2702 to be merged

Reply:
update